plugins/modules/ec2_vpc_endpoint_service.py

```diff
@@ -0,0 +1,513 @@
+            returned: always
+            type: bool
+        network_load_balancer_arns:
+            description: The Amazon Resource Names (ARNs) of the Network Load Balancers for the service.
+            returned: if network load balancers are backing the service
+            type: list
+        gateway_load_balancer_arns:
+            description: The Amazon Resource Names (ARNs) of the Gateway Load Balancers for the service.
+            returned: if gateway load balancers are backing the service
+            type: list
+        private_dns_name_configuration:
+            description: Information about the endpoint service private DNS name configuration.
+            returned: always
+            tpye: dict
+        service_id:
+            description: The ID of the service.
+            returned: always
+            type: str
+        service_name:
+            description: The name of the service.
+            returned: always
+            tpye: str
+        service_state:
+            description: The service state.
+            returned: always
+            tpye: str
+        service_type:
+            description: The type of service. Describes the type of service for a VPC endpoint.
+            returned: always
+            tpye: dict
+        supported_ip_address_types:
+            description: The supported IP address types.
+            returned: always
+            type: list
+        permissions:
+            description: The list of Amazon Resource Names (ARN) of the principals having access to the service.
+            returned: always
+            type: list
+        tags:
+            description: Tags of the endpoint service.
+            returned: always
+            type: dict
+"""
+
+try:
+    import botocore
+except ImportError:
+    pass  # Handled by AnsibleAWSModule
+
+import json
+from datetime import datetime
+from typing import Optional
+
+from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule
+from ansible_collections.amazon.aws.plugins.module_utils.core import is_boto3_error_code
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import AWSRetry
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import ansible_dict_to_boto3_filter_list
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import boto3_tag_list_to_ansible_dict
+from ansible.module_utils.common.dict_transformations import camel_dict_to_snake_dict
+from ansible.module_utils.common.dict_transformations import snake_dict_to_camel_dict
+
+from ansible.module_utils.resource_tags import ensure_tags
+from ansible_collections.amazon.aws.plugins.module_utils.tagging import ansible_dict_to_boto3_tag_list  # pylint: disable=unused-import
+
+try:
+    from botocore.exceptions import ClientError, BotoCoreError
+except ImportError:
+    pass  # Handled by AnsibleAWSModule
+
+ARGUMENT_SPEC = dict(
```

Comment:
I'd probably avoid setting defaults for most of the options here (except state).  While default behaviour on creation should be documented, when you're performing an update it's not possible to tell the difference between defaults and user-requested changes.  In general we should only change a parameter if someone explicitly passed that parameter.

Reply:
So the logic would be that there is no defaults set on module level, but we manage `None`s coming from user side when we need to create/update? (around lines 443 and 470)

Reply:
yes.

Reply:
> I'd probably avoid setting defaults for most of the options here (except state). While default behaviour on creation should be documented, when you're performing an update it's not possible to tell the difference between defaults and user-requested changes. In general we should only change a parameter if someone explicitly passed that parameter.

Is it ok if in the documentation and in ARGUMENT_SPEC I specify None as default for these parameters mentioning that defaults will be set as per boto3?

Reply:
That's fine (although it's the remote AWS service that'll set the default rather than the client SDK)

Reply:
updated